play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -179,36 +181,81 @@ export class ChatRoom extends PSRoom {
 			this.title = `[DM] ${nameWithGroup.trim()}`;
 		}
 	}
-	handleHighlight = (message: string, name: string) => {
-		if (!PS.prefs.noselfhighlight && PS.user.nameRegExp?.test(message)) {
-			this.notify({
-				title: `Mentioned by ${name} in ${this.id}`,
-				body: `"${message}"`,
-				id: 'highlight',
-			});
-			return true;
+	static getHighlight(message: string, roomid: string) {
+		let highlights = PS.prefs.highlights || {};
+		if (Array.isArray(highlights)) {
+			highlights = { global: highlights };
+			// Migrate from the old highlight system
+			PS.prefs.set('highlights', highlights);
+		}
+		if (!PS.prefs.noselfhighlight && PS.user.nameRegExp) {
+			if (PS.user.nameRegExp?.test(message)) return true;
 		}
-		/*
-		// TODO!
 		if (!this.highlightRegExp) {
 			try {
-				//this.updateHighlightRegExp(highlights);
-			} catch (e) {
+				this.updateHighlightRegExp(highlights);
+			} catch {
 				// If the expression above is not a regexp, we'll get here.
 				// Don't throw an exception because that would prevent the chat
 				// message from showing up, or, when the lobby is initialising,
 				// it will prevent the initialisation from completing.
 				return false;
 			}
 		}
-		var id = PS.server.id + '#' + this.id;
-		var globalHighlightsRegExp = this.highlightRegExp['global'];
-		var roomHighlightsRegExp = this.highlightRegExp[id];
-
-		return (((globalHighlightsRegExp &&
-		 globalHighlightsRegExp.test(message)) ||
-		  (roomHighlightsRegExp && roomHighlightsRegExp.test(message))));
-		*/
+		const id = PS.server.id + '#' + roomid;
+		const globalHighlightsRegExp = this.highlightRegExp?.['global'];
+		const roomHighlightsRegExp = this.highlightRegExp?.[id];
+		return (((globalHighlightsRegExp?.test(message)) || (roomHighlightsRegExp?.test(message))));
+	}
+	static updateHighlightRegExp(highlights: Record<string, string[]>) {
```

Comment:
Okay, actually this one shouldn't be static because it's called by commands. It's just the `highlightRegExp` itself that should be static. My bad.

Reply:
Sorry, no, I'm going back and forth on this. It's fine as-is.